containers/unit_tests/TestUnorderedMap.hpp

```diff
@@ -441,17 +441,49 @@ TEST(TEST_CATEGORY, UnorderedMap_valid_empty) {
   ASSERT_TRUE(n.is_allocated());
 }
 
+/**
+ * This helper is needed because NVCC does not like extended lambdas
+ * in private member functions.
+ * Google Test bodies are private member functions. So it is incompatible.
+ * See also https://github.com/google/googletest/issues/4104.
+ */
```

Comment:
Hi @dalg24 ! What do you think of this issue? Do you know the reason why Cuda requires that extended lambda cannot be defined in non-public members?

Reply:
It's just one of the restrictions of using lambda with CUDA https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#extended-lambda-restrictions